test/unit/transaction-runner-test.coffee

```diff
@@ -482,6 +482,484 @@ describe 'TransactionRunner', ()->
+            assert.equal spies.beforeEachSpy.callCount, 1
+
+          it 'should perform the before hook', () ->
+            assert.equal spies.beforeSpy.callCount, 1
+
+          it 'should perform the beforeEachValidation hook', () ->
+            assert.equal spies.beforeEachValidationSpy.callCount, 1
+
+          it 'should not perform any other hook', () ->
+            for spyName,spy of spies
+              break if spyName == 'beforeAllSpy'
+              break if spyName == 'beforeEachSpy'
+              break if spyName == 'beforeSpy'
+              break if spyName == 'beforeEachValidationSpy'
+
+              assert.isFalse spies[spyName].called, spyName
+
+          it 'should perform only the first transaction', () ->
+            assert.isTrue serverNock1.isDone(), 'first resource'
+            assert.isFalse serverNock2.isDone(), 'second resource'
+
+          it 'should return the error', () ->
+            assert.include returnedError.message, 'before each validation'
+
+        describe 'after beforeValidation hook is executed', () ->
+          beforeEach (done) ->
+            hookHandlerError = new Error 'handler died in before validation 1'
+
+            hooks.beforeValidation '1', (data, callback) ->
+              runner.hookHandlerError = hookHandlerError
+              callback()
+
+            runner.executeAllTransactions transactions, hooks, (error) ->
+              #setting expectation for this error below in each describe block
+              returnedError = error
+              done()
+
+          it 'should perform the beforeAll hook', () ->
+            assert.isTrue spies.beforeAllSpy.called
+
+          it 'should perform the beforeEach hook', () ->
+            assert.isTrue spies.beforeEachSpy.calledOnce
+
+          it 'should perform the before hook', () ->
+            assert.isTrue spies.beforeSpy.calledOnce
+
+          it 'should perform the beforeEachValidation hook', () ->
+            assert.isTrue spies.beforeEachValidationSpy.calledOnce
+
+          it 'should perform the beforeValidation', () ->
+            assert.isTrue spies.beforeValidationSpy.calledOnce
+
+          it 'should not perform any other hook', () ->
+            for spyName,spy of spies
+              break if spyName == 'beforeAllSpy'
+              break if spyName == 'beforeEachSpy'
+              break if spyName == 'beforeSpy'
+              break if spyName == 'beforeEachValidationSpy'
+              break if spyName == 'beforeValidationSpy'
+
+              assert.isFalse spies[spyName].called, spyName
+
+          it 'should perforom only first transaction', () ->
+            assert.isTrue serverNock1.isDone(), 'first resource'
+            assert.isFalse serverNock2.isDone(), 'second resource'
+
+          it 'should return the error', () ->
+            assert.include returnedError.message, 'before validation 1'
+
+        describe 'after afterEach hook is executed', () ->
```

Comment:
Order of the hook execution is incorrect, see [execution life cycle](http://dredd.readthedocs.org/en/latest/usage/#dredd-execution-lifecycle) `after` hook is exectued before `afterEach`. So the test cases should respect that.